compile/inc/src/main/scala/sbt/inc/IncrementalNameHashing.scala

```diff
@@ -18,7 +18,9 @@ private final class IncrementalNameHashing(log: Logger, options: IncOptions) ext
   // Package objects are fragile: if they inherit from an invalidated source, get "class file needed by package is missing" error
   //  This might be too conservative: we probably only need package objects for packages of invalidated sources.
   override protected def invalidatedPackageObjects(invalidated: Set[File], relations: Relations): Set[File] =
-    invalidated flatMap relations.inheritance.internal.reverse filter { _.getName == "package.scala" }
+    transitiveDeps(invalidated)(relations.inheritance.internal.reverse).filter {
+      _.getName == "package.scala"
+    }
```

Comment:
I realize it's not new to this commit, but what is the basis for this filter? It's not supposed to be a correctness requirement that package objects be defined in files called package.scala. It is (according to the scala specification) a convention.